### ISSUE TITLE
invite via email

### ISSUE THREAD

**Issue by yegor256** [yegor256/netbout#602]
It should be possible to invite a user into a bout, just by using his email. I just type in an email address into "invite" box and the user gets into the bout. He receives emails with all messages and is able to reply to them (after #600 is ready). When he clicks the link inside the email, he gets into web app and it asks him to register. Once registered, we link all bouts associated with this email to the newly registered URN.

**Comment by dmzaytsev** [comment]
@karato it's a valid bug

**Comment by dmzaytsev** [comment]
@karato set milestone to 3.0

**Comment by dmzaytsev** [comment]
@karato this task is posponed

**Comment by karato** [comment]
> @karato it's a valid bug
> 

@dmzaytsev tag "bug" added

**Comment by karato** [comment]
@yegor256 I added milestone `3.1` to this issue, let me know if there has to be something else

**Comment by karato** [comment]
> @karato set milestone to 3.0

@dmzaytsev thanks, I set it to "3.0"

**Comment by karato** [comment]
> @karato this task is postponed

@dmzaytsev got it, "postponed" tag here

**Comment by karato** [comment]
@komaz this task is for you, keep [these principles](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) in mind, and don't hesitate to ask any technical questions. The budget here is **30 mins**, which is exactly how much time will be paid for, when the task is [completed](http://www.xdsd.org/2014/04/15/definition-of-done.html)

**Comment by dmzaytsev** [comment]
@komaz let's use [GreenMail](http://www.icegreen.com/greenmail/) for email testing

**Comment by dmzaytsev** [comment]
@karato we are waiting #600 here

**Comment by karato** [comment]
@yegor256 thanks for this report, I added 15 mins to your account, in transaction `000-9b6df556`

**Comment by karato** [comment]
> @karato set milestone to 3.0

@dmzaytsev thanks, I set it to "3.0"

**Comment by karato** [comment]
> @karato this task is postponed

@dmzaytsev got it, "postponed" tag here

**Comment by karato** [comment]
> @karato this task is postponed

@dmzaytsev OK, I will try to assign someone else

**Comment by komaz** [comment]
@dmzaytsev could you please comment on a status of this issue, is it postponed or not? If it is waiting for #600, then I can't start working on that, right? Shouldn't it be unassigned from me to make sure I won't be penalised?

**Comment by dmzaytsev** [comment]
@komaz the task should not be assigned to anybody... 
Anyway just inform the project manager  http://www.teamed.io/policy.html#11

**Comment by komaz** [comment]
@karato waiting for #600

**Comment by karato** [comment]
> @karato waiting for #600 

@komaz sure, we'll wait for #600

**Comment by karato** [comment]
@lautarobock this ticket is yours now, please proceed, and keep in mind [this](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html). Any technical questions you should ask right here; The cost of this task is **30 mins** (this is exactly how much will be paid, not less not more), when the task is [done](http://www.xdsd.org/2014/04/15/definition-of-done.html)

**Comment by lautarobock** [comment]
@karato  #600 is not done yet.

**Comment by dmzaytsev** [comment]
@karato we are waiting #600 here

**Comment by lautarobock** [comment]
@karato Should not it be un-assigned? I won't be penalised

**Comment by dmzaytsev** [comment]
@karato this is postponed task

**Comment by karato** [comment]
> @karato this is postponed task

@dmzaytsev got it, "postponed" tag here

**Comment by karato** [comment]
> @karato this is postponed task

@dmzaytsev someone else will help in this task, no problem at all

**Comment by dmzaytsev** [comment]
@yegor256   do you mean send all existing bout messages to the entered email in single message?
> He receives emails with all messages and is able to reply to them

**Comment by yegor256** [comment]
@dmzaytsev no, no, just one by one. Every message I post to the bout, the user recipient receives to his email

**Comment by dmzaytsev** [comment]
@yegor256 I break this ticket on the three parts(1 task and two puzzle), please take a look

1. if the text entered into the invite box matched an email address regexp we should
сreate alias
  urn:    urn:email:<email>
  name: <email >  (replace @ to -)
  email:<email>

2. when a new message posted into bout we should add to an e-mail message for the email-user a text like this 'You are invited into the Netbout click on the link to register'
a link www.netbout.com/b/<bout_number>?invite=<invite-key> and 
where invite-key is encrypted urn e.g. we can use CcAES from Takes project

3. during the user registraton if the invite-key param presented in the original ticket we should 
shoud use alias with the urn equals the decrypted invite-key param.
then update the urn and the alias to the actual values

**Comment by dmzaytsev** [comment]
@karato this task not postponed any more

**Comment by karato** [comment]
> @karato this task not postponed any more

@dmzaytsev right, I removed "postponed" tag

**Comment by karato** [comment]
@lautarobock this task is yours, please help (see our key [principles of work](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html)). If you have any technical questions, don't hesitate to ask right here; The budget of this issue is **30 mins**, which is exactly how much will be paid when the task is *done* (see [this](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) for explanation)

**Comment by karato** [comment]
@yegor256 FYI, issue #600 is already closed, and we were waiting for it..

**Comment by yegor256** [comment]
@dmzaytsev looks like you're on the right track, let's proceed

**Comment by dmzaytsev** [comment]
@lautarobock  what's the status here?

**Comment by lautarobock** [comment]
@dmzaytsev I'm working on it now

**Comment by dmzaytsev** [comment]
@lautarobock please keep in mind http://www.teamed.io/policy.html#16

**Comment by dmzaytsev** [comment]
@lautarobock is there any progress here?

**Comment by karato** [comment]
@lautarobock the ticket is with you for 15 days already. If it is not finished in the next 48 hours, it will be re-assigned to someone else, see [No Obligations](http://www.xdsd.org/2014/04/13/no-obligations-principle.html) principle, [this article](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html) should help if you're stuck... -30 added to your rating, current score is: [-30](http://www.netbout.com/b/35344?open=rating)

**Comment by lautarobock** [comment]
@yegor256 @dmzaytsev about `urn: urn:email:<email>` is not possible. URN doesn't accept `@` nor `.`.
And with this specs, it have not way to detect **email users** for sending invitations

**Comment by dmzaytsev** [comment]
@lautarobock we don't need to store email in the urn as is. It just unique identifier...
let's use `some-hash-function(<email>)` instead `<email>` in the urn expression above

**Comment by lautarobock** [comment]
@dmzaytsev and how do you think to identify this kind of  **email users** ?

**Comment by dmzaytsev** [comment]
@lautarobock please see this comment https://github.com/yegor256/netbout/issues/602#issuecomment-110858137

we won't need to identify these email users, their urns will be updated to actual after they will be registered

**Comment by lautarobock** [comment]
@dmzaytsev ok, I thought  than in **every** mail has to be a link for invitation. sorry

**Comment by dmzaytsev** [comment]
@lautarobock you should not only create an alias, but also to invite it into the bout
I just updated https://github.com/yegor256/netbout/issues/602#issuecomment-110858137

**Comment by karato** [comment]
@lautarobock this task is taking too long, I have to change the performer, sorry. Please stop working with it right now. See our [no obligations](http://www.xdsd.org/2014/04/13/no-obligations-principle.html) principle

added -60 to your rating, now it is equal to [-150](http://www.netbout.com/b/35344?open=rating)

**Comment by dmzaytsev** [comment]
@karato please assign this task

**Comment by karato** [comment]
> @karato please assign this task

@dmzaytsev we don't have enough people in the project now... we're working on it now

**Comment by dmzaytsev** [comment]
@karato  thanks for letting me know

**Comment by dmzaytsev** [comment]
@antonini Could you pickup this ticket?
the action plan [here](https://github.com/yegor256/netbout/issues/602#issuecomment-110858137)

**Comment by karato** [comment]
> @karato  thanks for letting me know

@dmzaytsev np :)

**Comment by antonini** [comment]
@dmzaytsev All those three steps need to be implemented over this issue? Or should implement one and other steps are going to be puzzle?

**Comment by dmzaytsev** [comment]
@antonini time budget is 30min. Let''s start from first step. You should implement this one and create a puzzle for others

**Comment by dmzaytsev** [comment]
@karato assign this to @antonini please

**Comment by antonini** [comment]
@dmzaytsev Understood! I'll do that!

**Comment by karato** [comment]
> @karato assign this to @antonini please

@dmzaytsev yep, done. @antonini it is your task now

**Comment by antonini** [comment]
@dmzaytsev One thing. I can't create a URN like `urn:email:name@domain.com`, neither `urn:email:name-domain.com`. I'll replace `@` to `-` and also `.` to `-`. 
So final URN will be: `urn:email:name-domain-com`.
What you think about?

**Comment by dmzaytsev** [comment]
@antonini yes, you can use it.
we don't need to restore the real email from the urn. It should be just an unique identifier.
we can use even `urn:email:<some hash function from the email>`

**Comment by dmzaytsev** [comment]
@antonini it would be better to use a hash instead an email

**Comment by antonini** [comment]
@karato PR #746 ready for review. 

@dmzaytsev I've made some changes on the messages of Puzzles. Can you check it too?

**Comment by karato** [comment]
> @karato PR #746 ready for review. 
> 
> @dmzaytsev I've made some changes on the messages of Puzzles. Can you check it too?

@antonini thanks

**Comment by dmzaytsev** [comment]
@rultor deploy

**Comment by dmzaytsev** [comment]
@yegor256 @antonini I just invited a new user by email  into the bout on snapshop. When I posted a message into the bout it sent to the user's email as it requested in the ticket. I using http://10minutemail.com/ for temporary email address

**Comment by karato** [comment]
@yegor256 2 puzzles were created here: `602-3825bd3b`/#751, `602-8a4d4ea4`

**Comment by yegor256** [comment]
@dmzaytsev just checked it myself, works fine, thanks!

**Comment by karato** [comment]
@ypshenychka review this ticket please, for compliance with our [quality rules](http://www.teamed.io/qa.html)

**Comment by ypshenychka** [comment]
@karato Quality is good here.

**Comment by karato** [comment]
> @karato Quality is good here.

@ypshenychka thanks a lot

**Comment by karato** [comment]
@antonini 10 mins was added to the account of @ypshenychka (for QA review), in transaction 61729845... Many thanks! **30 mins** were added to your account in Transaction ID `AP-6CT28321N4834715A` (task took 178 hours and 16 mins)... added +30 to your rating, now it is equal to [+414](http://www.netbout.com/b/35346?open=rating)

**Comment by karato** [comment]
@yegor256 the last puzzle `751-6f10d03b`/#775 solved

**Comment by 0pdd** [comment]
@yegor256 all 3 puzzles are solved here: [#751](https://github.com/yegor256/netbout/issues/751), [#unknown](), [#unknown]().
